src/hackerspace_online/management/commands/full_clean.py

```diff
@@ -0,0 +1,58 @@
+from django.apps import apps
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+from django_tenants.utils import schema_context
+from tenant.models import Tenant
+
+
+class Command(BaseCommand):
+    LOCAL_APPS = (
+        'quest_manager',
+        'profile_manager',
+        'announcements',
+        'comments',
+        'notifications',
+        'courses',
+        'prerequisites',
+        'badges',
+        'djcytoscape',
+        'portfolios',
+        'utilities',
+        'siteconfig',
+        'tags',
+        'library',
+    )
+    help = "Loops through a list of tenants (all by default), and validates each object in the database using full_clean()"
+
+    def add_arguments(self, parser):
+
+        # optional arguments
+        parser.add_argument(
+            '--tenants', action='store', nargs='+',
+            help='A space separated list of tenant schemas. Specifies which tenant to loop through. Defaults to all tenants'
+        )
+
+    def handle(self, *args, **options):
+        tenant_names = options.get('tenants')
+
+        tenants = Tenant.objects.filter(schema_name__in=tenant_names) if tenant_names else Tenant.objects.all()
+
+        # querying models from these gives a relation error
+        tenants = tenants.exclude(schema_name__in=['library', 'public'])
+
+        # loop through all tenants
+        for tenant in tenants:
+            # loop through all models inside tenant
+            with schema_context(tenant.schema_name):
+
+                # loop through each model
+                for app_name in self.LOCAL_APPS:
+                    for model in apps.get_app_config(app_name).get_models():
+
+                        # full clean each object in model
+                        for object_ in model.objects.all().iterator(chunk_size=100):
+                            try:
+                                object_.full_clean()
+                                object_.save()
+                            except ValidationError as e:
+                                print(f'Exception found on cleaning {object_} ({model.__name__}):', e)
```

Comment:
Here's where we need some good debug output. In a code comment provide an example error (actually this should all be in the class's docstring explaining how to use this management command).

More importantly, what can we do about it?  Is the error informative enough?

Reply:
added some error examples to the docstring. also specified what kind of error shows up in the print statement

Since theres plenty of errors that full_clean may come across the print has to be generic enough to apply to all of them.
For example full clean can trigger a `IntegrityError`.